commit c0039918888e9e2d0f8edd284c52b3ffaabbef5b (refs/changes/08/2808/2)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2017-10-18T21:01:07+02:00 (2 years ago)
    
    SAPI-293 moray storage layer's putObject implementation doesn't handle errors correctly
    Reviewed by: Julien Gilli <julien.gilli@joyent.com>
    Approved by: Julien Gilli <julien.gilli@joyent.com>

diff --git a/lib/server/stor/moray.js b/lib/server/stor/moray.js
index aaded7d..cb26743 100644
--- a/lib/server/stor/moray.js
+++ b/lib/server/stor/moray.js
@@ -356,6 +356,7 @@ function putObject(bucket, uuid, obj, opts, cb) {
         if (err) {
             log.error(err, 'failed to put object %s', uuid);
             cb(err);
+            return;
         }
 
         var record = {};
